elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceTable.java

```diff
@@ -194,6 +281,26 @@ public void resolveAndStoreAllReferencesAndJoins(Queryable queryable) {
         return projections;
     }
 
+    private void addJoinContextsforCurrentContext(Queryable rootQueryable, TableContext tableCtx) {
+
+        Type<?> rootEntityClass = dictionary.getEntityClass(rootQueryable.getName(), rootQueryable.getVersion());
+
+        for (String joinField : dictionary.getFieldNamesWithAnnotations(rootEntityClass,
+                        Arrays.asList(Join.class, JoinColumn.class))) {
+            Type<?> joinClass = dictionary.getType(rootEntityClass, joinField);
```

Comment:
For a toMany join, this will return the collection type instead of the join type.  use getParameterizedType instead.